diff --git a/src/melee/ty/tydisplay.c b/src/melee/ty/tydisplay.c
index 5d71101..8ad9ad1 100644
--- a/src/melee/ty/tydisplay.c
+++ b/src/melee/ty/tydisplay.c
@@ -137,6 +137,11 @@ typedef struct TyDspConfig {
     /* 0x7C */ s32 x7C;
 } TyDspConfig;
 extern TyDspConfig* un_804D6F18;
+extern char un_803FF074[0xA8];
+
+char un_803FEFF0[0x2C] = "ToyDspPanel_Top_joint\0\0\0ToyDspBg_Top_joint";
+char un_803FF01C[0x2C] = "ToyDspStand_Top_joint\0\0\0ScMenDisplay_fog";
+static char un_804D5AA8[2] = "0";
 
 void un_803182D4_OnFrame(void)
 {
@@ -997,7 +1002,7 @@ void un_80319EF0(void)
             }
         }
         if (scale > 2.1474836e9f || scale < -2.1474836e9f) {
-            OSReport("*** tyDisplay Table Scale Irregul!\n");
+            OSReport(un_803FF074);
             HSD_ASSERT(0x28C, 0);
         }
         if ((s32) scale != 0) {
@@ -1373,6 +1378,7 @@ void fn_8031A94C(HSD_GObj* arg0)
     }
 }
 
+char un_803FF074[0xA8] = "*** tyDisplay Table Scale Irregul!\n\0TyMnDisp.dat\0\0\0\0TyMnDisp.usd\0\0\0\0*** BG data aren't being loaded!\n\0\0\0*** Can not Load Panel Label(%s)\n\0\0\0ScMenDisplay_scene_lights";
 static u16 un_804D5ABC = 0x15;
 
 void un_8031B1FC(void)
@@ -1384,6 +1390,7 @@ void un_8031B1FC(void)
     int zero;
     u8 temp;
     HSD_JObj* jobj;
+    char* strbase = un_803FEFF0;
     gobj4 = ptr->gobj4;
     zero = 0;
     do {
@@ -1391,8 +1398,8 @@ void un_8031B1FC(void)
     } while (zero);
 
     if (ptr->archive == NULL) {
-        OSReport("*** BG data aren't being loaded!\n");
-        HSD_ASSERT(0x3FD, 0);
+        OSReport(strbase + 0xC8);
+        __assert(strbase + 0x78, 0x3FD, un_804D5AA8);
     }
 
     gobj = ptr->gobj0;
@@ -1409,7 +1416,7 @@ void un_8031B1FC(void)
         ptr->gobj4 = NULL;
     }
 
-    joint = HSD_ArchiveGetPublicAddress(ptr->archive, "ToyDspBg_Top_joint");
+    joint = HSD_ArchiveGetPublicAddress(ptr->archive, strbase + 0x18);
     if (joint != NULL) {
         ptr->gobj4 = GObj_Create(9, 9, zero);
         jobj = HSD_JObjLoadJoint(joint);
@@ -1419,8 +1426,8 @@ void un_8031B1FC(void)
         return;
     }
 
-    OSReport("*** Can not Load Panel Label(%s)\n", "ToyDspBg_Top_joint");
-    HSD_ASSERT(0x43E, 0);
+    OSReport(strbase + 0xEC, strbase + 0x18);
+    __assert(strbase + 0x78, 0x43E, un_804D5AA8);
 }
 
 static s32 un_804DE018 = (s32) 0xC8C8C8FF;
@@ -1428,10 +1435,11 @@ static f32 un_804DE01C = 0.6f;
 
 void un_8031B328(void)
 {
-    HSD_FogDesc* fogDesc;
+    char* strbase = un_803FEFF0;
     TyDspBgData* ptr = un_804D6F1C;
     TyDspSceneGfx* scene = (TyDspSceneGfx*) un_804D6ED4;
     LightList** lightData;
+    HSD_FogDesc* fogDesc;
     TyDspBgData* temp3;
     s8 temp2;
     HSD_LObj* lobj;
@@ -1440,13 +1448,13 @@ void un_8031B328(void)
     PAD_STACK(24);
 
     if ((temp3 = ptr)->archive == NULL) {
-        OSReport("*** BG data aren't being loaded!\n");
-        OSPanic(__FILE__, 0x459, "0");
+        OSReport(strbase + 0xC8);
+        OSPanic(strbase + 0x78, 0x459, "");
     }
 
-    lightData = HSD_ArchiveGetPublicAddress(temp3->archive,
-                                            "ScMenDisplay_scene_lights");
-    if (lightData != NULL) {
+    if ((lightData = HSD_ArchiveGetPublicAddress(temp3->archive,
+                                                 strbase + 0x110)) != NULL)
+    {
         scene->x00 = GObj_Create(2, 3, 0);
         lobj = Toy_LoadLObjList(lightData, 0);
         HSD_GObjObject_80390A70(scene->x00, (unsigned long) HSD_GObj_804D784A,
@@ -1454,14 +1462,13 @@ void un_8031B328(void)
         temp = scene->x00;
         GObj_SetupGXLink(temp, HSD_GObj_LObjCallback, 0x34, 0);
     }
-    if ((((un_804D6F20 != 0) & 0xFFFFFFFFFFFFFFFF) & 0xFFFFFFFFFFFFFFFF) &
-        0xFFFFFFFFFFFFFFFF)
-    {
+    if (un_804D6F20 != 0) {
         HSD_LObjSetColor(lobj, *(GXColor*) &un_804DE018);
     }
 
-    fogDesc = HSD_ArchiveGetPublicAddress(temp3->archive, "ScMenDisplay_fog");
-    if (fogDesc != NULL) {
+    if ((fogDesc = HSD_ArchiveGetPublicAddress(temp3->archive, strbase + 0x44)) !=
+        NULL)
+    {
         scene->x08 = GObj_Create(3, 4, 0);
         HSD_GObjObject_80390A70(scene->x08, temp2 = HSD_GObj_804D7848,
                                 HSD_FogLoadDesc(fogDesc));
@@ -1469,8 +1476,6 @@ void un_8031B328(void)
     }
 }
 
-static char un_803FEFF0[] = "ToyDspPanel_Top_joint";
-
 void un_8031B460_OnEnter(void* arg0)
 {
     s32 sp18;
@@ -1972,6 +1977,11 @@ HSD_GObj* un_8031BC54(s32 arg0)
 static char un_804D5AAC[] = "jobj.h";
 static char un_804D5AB4[] = "jobj";
 
+static inline HSD_JObj* un_8031BF34_inline(void)
+{
+    return (HSD_JObj*) un_804D6F2C->hsd_obj;
+}
+
 void un_8031BF34(s32 arg0)
 {
     TyDspBaseData* base = &un_804A2D98;
@@ -1982,21 +1992,20 @@ void un_8031BF34(s32 arg0)
 
     if (un_804D6F2C != NULL) {
         HSD_Archive** archp = &base->archive;
-        HSD_Archive* arch = *archp;
-        if (arch != NULL) {
-            lbArchive_80016EFC(arch);
+        if (base->archive != NULL) {
+            lbArchive_80016EFC(base->archive);
             *archp = NULL;
         }
     }
 
-    un_80308250(base->x38, (s16) arg0, 0);
+    un_80308250(base->x38, (s16) (u16) arg0, 0);
     un_804D6F2C = un_803087F4(base->x38);
 
     HSD_JObjClearFlagsAll(anim->jobj[0], 0x10);
     HSD_JObjSetFlagsAll(anim->jobj[1], 0x10);
     HSD_JObjClearFlagsAll(anim->jobj[0], 0x10);
 
-    jobj = (HSD_JObj*) un_804D6F2C->hsd_obj;
+    jobj = un_8031BF34_inline();
 
     HSD_JObjSetScaleX(jobj, 0.6f);
     HSD_JObjSetScaleY(jobj, 0.6f);
